src/helpers.c

```diff
@@ -671,6 +660,14 @@ void string_or_die(char **strp, const char *fmt, ...)
 	va_end(ap);
 }
 
+void free_string(char **s)
```

Comment:
The reason for checking that s is not null is just to stop cores. It is a programming error if s is null. So it might be worth declaring this as [nonnull](https://gcc.gnu.org/onlinedocs/gcc-4.7.0/gcc/Function-Attributes.html#index-g_t_0040code_007bnonnull_007d-function-attribute-2555)

Reply:
And remove the check in free_string()?

Reply:
No, they serve different purposes. The check in free_string is to avoid a runtime crash. The annotation enables a compile time warning (which we turn into a compile time error).

The warning probably will never be triggered, but defense in depth is not a bad thing.

Reply:
OK, but when I add nonnull the compiler complains that free_string itself is comparing a nonnull value to null.

Reply:
+1 for option 1, this is good to go then?

Reply:
Yes